fearless/Common/Services/ExtrinsicService/ExtrinsicService.swift

```diff
@@ -165,33 +280,54 @@ extension ExtrinsicOperationFactory: ExtrinsicOperationFactoryProtocol {
             dependingOn: nonceOperation,
             codingFactoryOperation: codingFactoryOperation,
             customClosure: closure,
+            numberOfExtrinsics: numberOfExtrinsics,
             signingClosure: signingClosure
         )
 
         builderOperation.addDependency(nonceOperation)
         builderOperation.addDependency(codingFactoryOperation)
 
-        let submitOperation = JSONRPCListOperation<String>(
-            engine: engine,
-            method: RPCMethod.submitExtrinsic
-        )
-        submitOperation.configurationBlock = {
-            do {
-                let extrinsic = try builderOperation
-                    .extractNoCancellableResultData()
-                    .toHex(includePrefix: true)
-
-                submitOperation.parameters = [extrinsic]
-            } catch {
-                submitOperation.result = .failure(error)
+        let submitOperationList: [JSONRPCListOperation<String>] =
+            (0 ..< numberOfExtrinsics).map { index in
+                let submitOperation = JSONRPCListOperation<String>(
+                    engine: engine,
+                    method: RPCMethod.submitExtrinsic
+                )
+
+                submitOperation.configurationBlock = {
+                    do {
+                        let extrinsic = try builderOperation
+                            .extractNoCancellableResultData()[index]
+                            .toHex(includePrefix: true)
+
+                        submitOperation.parameters = [extrinsic]
+                    } catch {
+                        submitOperation.result = .failure(error)
+                    }
+                }
+
+                submitOperation.addDependency(builderOperation)
+
+                return submitOperation
+            }
+
+        let wrapperOperation = ClosureOperation<[SubmitExtrinsicResult]> {
+            submitOperationList.map { submitOperation in
+                if let result = submitOperation.result {
+                    return result
+                } else {
+                    return .failure(BaseOperationError.parentOperationCancelled)
+                }
             }
         }
 
-        submitOperation.addDependency(builderOperation)
+        submitOperationList.forEach { submitOperation in
+            wrapperOperation.addDependency(submitOperation)
+        }
 
         return CompoundOperationWrapper(
-            targetOperation: submitOperation,
-            dependencies: [nonceOperation, codingFactoryOperation, builderOperation]
+            targetOperation: wrapperOperation,
+            dependencies: [nonceOperation, codingFactoryOperation, builderOperation] + wrapperOperation.dependencies
         )
```

Comment:
Please, put manually feeOperationList otherwise we can at some point add an operation twice